src/java.base/share/classes/java/lang/PublicMethods.java

```diff
@@ -153,6 +153,10 @@ static MethodList filter(Method[] methods, String name,
                     Key.matches(method, name, ptypes)) {
                     if (tail == null) {
                         head = tail = new MethodList(method);
+                        if (ptypes.length == 0) {
+                            // zero args can only have one match - stop looking
```

Comment:
This is wrong: if superclass has `Object work()` and this class has `Integer work()`, bytecode for this class will have a synthetic final `Object work()` that delegates call to `Integer work()`.

Reply:
Gotcha, I'll remove the no-arg change.  Thanks!